client/components/Main/FilterSubnavDropdown.tsx

```diff
@@ -0,0 +1,266 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+
+import {superdeskApi, planningApi} from '../../superdeskApi';
+import {FILTER_TYPE, IAgenda, ICalendar, ISearchFilter} from '../../interfaces';
+
+import {AGENDA, EVENTS, EVENTS_PLANNING, MAIN} from '../../constants';
+
+import {activeFilter, currentSearchFilterId} from '../../selectors/main';
+import {calendars, currentCalendarId} from '../../selectors/events';
+import {agendas, currentAgendaId} from '../../selectors/planning';
+import {combinedViewFilters} from '../../selectors/eventsplanning';
+
+import {Dropdown} from '../UI/SubNav';
+
+interface IDropdownItem {
+    id?: string;
+    label?: string;
+    action?(): void;
+    disabled?: boolean;
+    icon?: string;
+    divider?: boolean;
+}
```

Comment:
would probably go rather for something like:
```ts
interface IDropdownAction {
    id: string;
    label: string;
    action: () => void;
    ...
}

interface IDropdownDivider {
    divider: true;
}

type IDropdownItem = IDropdownAction | IDropdownDivider;
```

so it wouldn't need everything optional